internal/engine/actions/actions.go

```diff
@@ -138,22 +138,82 @@ func (rae *RuleActionsEngine) processAction(
 	params engif.ActionsParams,
 	metadata *json.RawMessage,
 ) (json.RawMessage, error) {
+	logger := zerolog.Ctx(ctx).Debug().
+		Str("action", string(actionType)).
+		Str("cmd", string(cmd))
+
+	// Exit early in case we shouldn't do anything, i.e., cmd != ActionCmdOn or cmd != ActionCmdOff
+	if cmd == engif.ActionCmdDoNothing {
+		logger.Msg("skipping action")
+		switch actionType {
+		case remediate.ActionType:
+			// Return the previous remediation status.
+			err := enginerr.RemediationStatusAsError(params.GetEvalStatusFromDb().RemStatus.RemediationStatusTypes)
+			return *(getMeta(params.GetEvalStatusFromDb().RemMetadata)), err
+		case alert.ActionType:
+			// Return the previous alert status.
+			err := enginerr.AlertStatusAsError(params.GetEvalStatusFromDb().AlertStatus.AlertStatusTypes)
+			return *(getMeta(params.GetEvalStatusFromDb().AlertMetadata)), err
+		}
+	}
 	// Get action engine
 	action := rae.actions[actionType]
 	// Return the result of the action
-	logger := zerolog.Ctx(ctx)
-	logger.Debug().
-		Str("action", string(actionType)).
-		Str("cmd", string(cmd)).
-		Msg("invoking action")
+	logger.Msg("invoking action")
 	return action.Do(ctx, cmd, rae.actionsOnOff[actionType], ent, params, metadata)
 }
 
 // shouldRemediate returns the action command for remediation taking into account previous evaluations
 func shouldRemediate(prevEvalFromDb *db.ListRuleEvaluationsByProfileIdRow, evalErr error) engif.ActionCmd {
-	_ = prevEvalFromDb
-	_ = evalErr
-	return engif.ActionCmdOn
+	// Get current evaluation status
+	newEval := enginerr.ErrorAsEvalStatus(evalErr)
+
+	// Get previous evaluation status
+	prevEval := db.EvalStatusTypesPending
+	if prevEvalFromDb.EvalStatus.Valid {
+		prevEval = prevEvalFromDb.EvalStatus.EvalStatusTypes
+	}
+
+	// Get previous Remediation status
+	prevRemediation := db.RemediationStatusTypesSkipped
+	if prevEvalFromDb.AlertStatus.Valid {
+		prevRemediation = prevEvalFromDb.RemStatus.RemediationStatusTypes
+	}
+
+	// Start evaluation scenarios
+
+	// Case 1 - Do nothing if the evaluation status has not changed and remediation did not errored-out
+	if newEval == prevEval && prevRemediation != db.RemediationStatusTypesError {
+		return engif.ActionCmdDoNothing
+	}
```

Comment:
aren't there legitimate cases for the PR evaluator where we want to push a new version of the PR even if the evaluation was failure before and is still failing?
e.g. I want frizbee to unroll tags to shas. The evaluation is failing and there is a PR. While the PR is open, someone pushes another workflow with more actions..

Reply:
That's a good point, I haven't thought about the frizbee action as I assumed we are proposing only new files so far (dependabot, codeql, trivy, etc). I'll try to think about how to implement that 👍 Do you know if it's working like that right now and this will break it or?

Reply:
with frizbee in particular I think the additional changes are the way it works now, yes, because unllike dependabot and codeql where we just put in the content, with frizbee you feed the content you get from evaluation to the frizbee library which modifies it and gives you back the modified content.

Reply:
I'm fine fixing this in a follow-up, but could this be solved by:
 - if the old status is failure always defer to the remediator
 - in the PR remediator, check what HEAD the PR points to. If it is the same as the hash of the commit we created locally, shortcut, otherwise go forward and force-push
 - the other remediators we have at least for now are "atomic" so this should not happen in the first place and if it does, then we should have anotoher mechanism that prevents an infinite remediate loop anyway